### PR TITLE
Issue #506 Bring in flowcrypt-mobile-core

### LINKED ISSUE #506: bring flowcrypt-mobile-core repo contents into flowcrypt-ios repo
There is a repository https://github.com/FlowCrypt/flowcrypt-mobile-core where we had OpenPGP.js based tooling (called "Core") for both Android and iOS.

On Android we already transitioned to PGPainless based implementation, so that repository is now only used by iOS.

I'd like to bring the contents into this repository, to make it easier to maintain going forward. 

1) I propose a folder called `Core` ? Then inside of that, there would be most things you see here: https://github.com/FlowCrypt/flowcrypt-mobile-core
2) the tests that run here https://github.com/FlowCrypt/flowcrypt-mobile-core/blob/master/.semaphore/semaphore.yml should be added to semaphore file on this iOS repository
3) you don't have to update other `scripts` in package.json - I need to clean these up myself
4) the result of the build does not need to be copied into the iOS Swift project (to use the built file, for now we'll have to manually copy it from the built result into https://github.com/FlowCrypt/flowcrypt-ios/blob/master/FlowCrypt/Resources/flowcrypt-ios-prod.js.txt

### PR DESCRIPTION
**Pull request by IvanPizhenko**
This PR brings in contents of the https://github.com/FlowCrypt/flowcrypt-mobile-core into this repo.

close #506

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: Core/LICENSE [hunk 1]
```diff
@@ -0,0 +1,11 @@
+Owner: FlowCrypt Limited
```
Comment (tomholub) [inline Core/LICENSE:1]: Please remove the `LICENSE` file, it will go by the main `LICENSE` file in the repo.

File: Core/package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "updateCore": "npm run-script _updateCoreLib && npm run-script _updateCoreSrc",
+    "_updateCoreSrc": "cp -r ../flowcrypt-browser/extension/js/common/core source/core",
+    "_updateCoreLib": "mkdir source/lib && cd source/lib && LIBD='../../../flowcrypt-browser/extension/lib' && cp -r $LIBD/emailjs . && cp $LIBD/openpgp.js . && cp $LIBD/iso-*.js . && cp $LIBD/zxcvbn.js .",
+    "donateCore": "rm -rf ../flowcrypt-browser/extension/js/common/core && cp -r source/core ../flowcrypt-browser/extension/js/common/core"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/FlowCrypt/flowcrypt-mobile-core.git"
+  },
+  "author": "FlowCrypt Limited",
```
Comment (tomholub) [inline Core/package.json:42]: Please update to `FlowCrypt a. s.`

File: Core/package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    "preupdateCore": "rm -rf source/core source/lib",
+    "updateCore": "npm run-script _updateCoreLib && npm run-script _updateCoreSrc",
+    "_updateCoreSrc": "cp -r ../flowcrypt-browser/extension/js/common/core source/core",
+    "_updateCoreLib": "mkdir source/lib && cd source/lib && LIBD='../../../flowcrypt-browser/extension/lib' && cp -r $LIBD/emailjs . && cp $LIBD/openpgp.js . && cp $LIBD/iso-*.js . && cp $LIBD/zxcvbn.js .",
+    "donateCore": "rm -rf ../flowcrypt-browser/extension/js/common/core && cp -r source/core ../flowcrypt-browser/extension/js/common/core"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/FlowCrypt/flowcrypt-mobile-core.git"
+  },
```
Comment (tomholub) [inline Core/package.json:41]: I suppose this can be removed - thanks!

File: Core/package.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  "author": "FlowCrypt Limited",
+  "license": "SEE LICENSE IN <LICENSE>",
+  "private": true,
+  "bugs": {
+    "url": "https://github.com/FlowCrypt/flowcrypt-mobile-core/issues"
+  },
```
Comment (tomholub) [inline Core/package.json:47]: Probably this could be removed too

File: Core/package.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "flowcrypt-mobile-core",
+  "version": "0.0.1",
+  "description": "Node.js bundle for FlowCrypt Android internal use.",
```
Comment (tomholub) [inline Core/package.json:4]: Let's update to `TypeScript core for FlowCrypt iOS internal use`

**Review by tomholub — COMMENTED** [review]
Looks good - noticed a few details

**Comment by IvanPizhenko** [comment]
@tomholub I've implemented above changes. Please recheck & approve.

File: .semaphore/semaphore.yml [hunk 6]
```diff
@@ -23,9 +23,13 @@ blocks:
...
-            - cd ~/git/flowcrypt-ios/Core
-            - npm install
-            - npm test
+            - mkdir .custom-npm
+            - cd .custom-npm
+            - npm install npm@7.23.0
+            - cd ../Core
+            - ../.custom-npm/node_modules/.bin/npm --version
+            - ../.custom-npm/node_modules/.bin/npm install
+            - ../.custom-npm/node_modules/.bin/npm test
```
Comment (tomholub) [inline .semaphore/semaphore.yml]: I believe you can do sem-version npm 7.23.0 but I'm not 100% sure. Anyway if yes, that can be fixed later.

**Review by tomholub — APPROVED** [review]
Excellent, thanks!

**Comment by tomholub** [comment]
I've re-arranged them to a single block from this:

image

To this:

image

I figure, they should run faster that way (althrough there is a quota for 2 simultaneous tasks on apple hardware, I've just requested a quota increase)

**Review by tomholub — APPROVED** [review]

**Comment by tomholub** [comment]
>  is documented clearly and usefully, or doesn't need documentation

This checkbox made me think - later we'll document this in readme. I'll mention it in the next issue.

**Pull request merged**